Assets/Scripts/UI/Text/TextController.cs

```diff
@@ -44,14 +37,13 @@ public void Update()
         {
             ResetTexts();
```

Comment:
You don't have to reset the text here. It creates more problems and erases the stats of the player from the UI which is arguably bad